samples/Makefile

```diff
@@ -106,73 +69,9 @@ ifneq (,$(findstring m68k,$(UNICORN_ARCHS)))
 SOURCES += sample_m68k.c
 endif
 
-OBJS = $(addprefix $(OBJDIR)/,$(SOURCES:.c=.o))
-OBJS_ELF = $(addprefix $(OBJDIR)/,$(SOURCES:.c=))
-BINARY = $(addprefix $(SAMPLEDIR)/,$(SOURCES:.c=$(BIN_EXT)))
-
-all: $(BINARY)
-
-clean_bins:
-	rm -rf *.o $(OBJS_ELF) $(BINARY) $(SAMPLEDIR)/*.exe $(SAMPLEDIR)/*.static $(OBJDIR)/lib$(LIBNAME)* $(OBJDIR)/$(LIBNAME)*
-	rm -rf sample_x86 sample_arm sample_arm64 sample_mips sample_sparc sample_ppc sample_m68k shellcode mem_apis sample_x86_32_gdt_and_seg_regs sample_batch_reg
-
-clean_libs:
-	rm -rf libunicorn*.so libunicorn*.lib libunicorn*.dylib unicorn*.dll unicorn*.lib
-
-clean: clean_bins clean_libs
-
-$(BINARY): $(OBJS)
-
-$(SAMPLEDIR)/%$(BIN_EXT): $(OBJDIR)/%.o
-	@mkdir -p $(@D)
-ifeq ($(V),0)
-ifeq ($(UNICORN_SHARED),yes)
```

Comment:
i don't think there's any reason to compile and link in two separate steps for what's being done here.

also, there's a huge advantage following GNU standards and using implicit rules. it helps with library and flag ordering, etc